packages/addon/src/channel_selector.ts

```diff
@@ -35,40 +40,95 @@ function fillChannels(data: ChannelDetails[], selected: string | null, messageCl
 
 window.addEventListener("load", () => {
   const parent = window.parent;
-  const logo = document.getElementById("logo")!!
+  const logo = document.getElementById("logo")!;
 
   const mc = new MessageChannel();
-  const myPort = mc.port1
-  myPort.start()
-
-  parent.postMessage({ type: "SelectorMessageInitialize" }, "*", [mc.port2]);
-
-  function changeSize(expanded: boolean) {
-    document.body.setAttribute("data-expanded", "" + expanded);
-    myPort.postMessage({ type: "SelectorMessageResize", expanded })
-  }
-
-  myPort.addEventListener("message", (e: MessageEvent) => {
-    if (e.data.type == 'SelectorMessageChannels') {
-      const details = e.data as SelectorMessageChannels
-      // console.log(JSON.stringify("CHANNEL DETAILS: " + JSON.stringify(details)))
-      channels = details.channels
-      channelId = details.selected
-
-      const selectedColor = (channelId ? (channels.find(c => c.id == channelId)?.displayMetadata?.color) : null) ?? 'black'
-      logo.style.fill = selectedColor
+  const myPort = mc.port1;
+  myPort.start();
+  myPort.onmessage = ({data}) => {
+    console.debug("Received message: ", data);
+    switch(data.type){
+      case "iframeHandshake": {
+        collapse();
+        break;
+      }
+      case "iframeChannels": {
+        logo.removeEventListener("click", expand);
+        const {userChannels, selected} = data.payload as IframeChannelsPayload;
```

Comment:
IframeChannelsPayload needs updating to the new type name. 

```suggestion
        const {userChannels, selected} = data.payload as Fdc3UserInterfaceChannelsPayload;
```
However, you are also using one of the generated components of the top level type. The naming of these is unstable and they should not be used directly. Instead, you should type the `data` element as `Fdc3UserInterfaceChannels` instead and rework accordingly